### PR TITLE
Framework: Deprecate use of component-classes

### PR DESCRIPTION
**Pull request by furf**
* Replaced use of `component-classes` methods with native JavaScript `classList` API methods.
* Removed (mockery) references to `component-classes` from test scripts.
* Removed mock file for `component-classes` that did not appear to be referenced in the codebase.
* Removed reference to `component-classes` module from server/bundler/loader

/cc @apeatling @aduth @gziolo @umurkontaci @nb

### CODE REVIEW

**Comment by aduth** [comment]
The [test helpers documentation](https://github.com/Automattic/wp-calypso/tree/remove/5363-component-classes/test/test/helpers/mocks) still includes an example referencing the `component-classes` mock helper.

Should we also remove it from [our dependencies](https://github.com/Automattic/wp-calypso/blob/remove/5363-component-classes/package.json#L30) ? (including shrinkwrap regeneration)

**Comment by furf** [comment]
I didn't know if it was important to remove the reference from the documentation as the instruction describes general implementation. I can update it to use `data-poller` since `component-tip` is also slated for the chopping block.

Yes. I believe it can be removed from the project's dependencies.

Thanks for the feedback. PR has been updated with commits reflecting these changes.

File: package.json [hunk 1]
```diff
@@ -27,7 +27,6 @@
     "click-outside": "1.0.4",
     "clipboard": "1.5.3",
     "commander": "2.3.0",
-    "component-classes": "1.2.6",
```
Comment (gziolo) [inline package.json]: 👍

**Comment by aduth** [comment]
@furf : #5404, #5434, and this branch can (and should) still be rebased, even though you've already merged changes from the master branch. Rebasing (and squashing commits like 27d6ca004b1d1ed870035c4454120c89e9016dd1) is part of [our recommended git workflow](https://github.com/Automattic/wp-calypso/blob/master/docs/git-workflow.md#keeping-your-branch-up-to-date).

I believe when you rebase after having merged in another branch, that merge commit will automatically be excluded, though you can confirm this by running your rebase interactively (`git rebase -i origin/master`) and verifying the picked commits.

**Comment by furf** [comment]
Thanks for the advice. I rebased my 3 PR's and the merge commits were removed.

File: server/bundler/loader.js [hunk 2]
```diff
@@ -12,7 +12,6 @@ function getSectionsModule( sections ) {
 			"\tlayoutFocus = require( 'lib/layout-focus' ),",
 			"\tReact = require( 'react' ),",
 			"\tLoadingError = require( 'layout/error' ),",
-			"\tclasses = require( 'component-classes' ),",
```
Comment (aduth) [inline server/bundler/loader.js]: Curious why this didn't require more changes, but looks like the dependency was left erroneously (6224-gh-calypso-pre-oss).

**Comment by aduth** [comment]
This looks good to me 👍

**Pull request merged**